sparse/impl/KokkosSparse_spiluk_symbolic_impl.hpp

```diff
@@ -147,7 +147,8 @@ void level_sched_tp(IlukHandle& thandle, const RowMapType row_map, const Entries
   if (std::is_same<memory_space, Kokkos::CudaSpace>::value) {
     size_t free_byte, total_byte;
     KokkosKernels::Impl::kk_get_free_total_memory<memory_space>(free_byte, total_byte);
-    avail_byte = static_cast<size_t>(0.85 * static_cast<double>(free_byte) / static_cast<double>(nstreams));
+    double orig_matrix_bytes = entries.extent(0) * 16;
+    avail_byte = static_cast<size_t>(std::min(0.85 * static_cast<double>(free_byte), orig_matrix_bytes) / static_cast<double>(nstreams));
```

Comment:
@vbrunini I am just concerned that when we have a matrix with a few nnz so ```orig_matrix_bytes``` is always smaller than ```free_byte```, we will end up having a lot of chunks which may slows down the spiluk numeric. Though I have not thought of an estimation better than using ```orig_matrix_bytes```.  Have you seen any performance change on your side with this change?

Reply:
I see you pick a floor of 128MB and I am fine with it. This PR looks good to me for now. Next FY, I will probably look into performance optimization for spiluk more carefully.